### PR TITLE
get rid of `window.interceptor.interceptorAccessDialog` and use bigints for addresses instead of mixing them

### PR DESCRIPTION
**Pull request by KillariDev**

### CODE REVIEW

File: extension/app/ts/background/windows/interceptorAccess.ts [hunk 1]
```diff
@@ -1,44 +1,67 @@
...
+		origin: pendingInterceptorAccess.origin,
+		requestAccessToAddress: pendingInterceptorAccess.requestAccessToAddress
+	})
+}
+
 export async function resolveInterceptorAccess(confirmation: Confirmation) {
-	if (pendingInterceptorAccess !== undefined) pendingInterceptorAccess.resolve(confirmation)
+	if (pendingInterceptorAccess !== undefined)
+	if (confirmation.origin === pendingInterceptorAccess.origin && confirmation.requestAccessToAddress === pendingInterceptorAccess.requestAccessToAddress)
+	pendingInterceptorAccess.future.resolve(confirmation)
```
Comment (MicahZoltu) [inline extension/app/ts/background/windows/interceptorAccess.ts:49]:
```suggestion
	if (pendingInterceptorAccess !== undefined) {
		if (confirmation.origin === pendingInterceptorAccess.origin && confirmation.requestAccessToAddress === pendingInterceptorAccess.requestAccessToAddress) {
			pendingInterceptorAccess.future.resolve(confirmation)
		}
	}
```
```suggestion
	if (pendingInterceptorAccess !== undefined
		&& confirmation.origin === pendingInterceptorAccess.origin
		&& confirmation.requestAccessToAddress === pendingInterceptorAccess.requestAccessToAddress
	) {
		pendingInterceptorAccess.future.resolve(confirmation)
	}
```
```suggestion
	if (pendingInterceptorAccess !== undefined && confirmation.origin === pendingInterceptorAccess.origin && confirmation.requestAccessToAddress === pendingInterceptorAccess.requestAccessToAddress) pendingInterceptorAccess.future.resolve(confirmation)
```

File: extension/app/ts/components/pages/InterceptorAccess.tsx [hunk 2]
```diff
@@ -16,50 +17,34 @@ export function InterceptorAccess() {
 	const [addressBookEntryInput, setAddressBookEntryInput] = useState<AddressBookEntry | undefined>(undefined)
 
 	useEffect( () => {
-		function popupMessageListener(msg: unknown) {
-			console.log('popup message')
-			console.log(msg)
-			fetchAccessDialog()
+		async function popupMessageListener(msg: unknown) {
+			const message = MessageToPopup.parse(msg)
+			if ( message.message !== 'popup_interceptorAccessDialog') return
```
Comment (MicahZoltu) [inline extension/app/ts/components/pages/InterceptorAccess.tsx:22]:
```suggestion
			if (message.message !== 'popup_interceptorAccessDialog') return
```

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
Blocked on fixing the non-obvious nested conditional.

File: extension/app/ts/background/accessManagement.ts [hunk 3]
```diff
@@ -277,10 +278,10 @@ export function updateWebsiteApprovalAccesses() {
 		const websiteAccess = hasAccess(window.interceptor.settings.websiteAccess, connection.origin)
 		if (activeAddress) {
 			// check for address access changes
-			const addressAccess = hasAddressAccess(window.interceptor.settings.websiteAccess, connection.origin, addressString(activeAddress))
+			const addressAccess = hasAddressAccess(window.interceptor.settings.websiteAccess, connection.origin, activeAddress)
 
 			if (addressAccess === 'notFound') {
-				askUserForAccessOnConnectionUpdate(port, connection.origin, addressString(activeAddress))
+				askUserForAccessOnConnectionUpdate(port, connection.origin, findAddressInfo(activeAddress, window.interceptor.settings.addressInfos) )
```
[the remark this answers is not in the document]
Comment (KillariDev) [reply extension/app/ts/background/accessManagement.ts:284]: It is, but I am not doing it everything at once as that would make a very big PR at once with a lot bugs. I have allowed global address database, its going to be yet another PR when I remove that :)

File: extension/app/ts/background/windows/interceptorAccess.ts (hunk 1 above)
Comment (KillariDev) [reply extension/app/ts/background/windows/interceptorAccess.ts:49]: these should be guard clauses, I am clearly been too tired when I wrote this, the first line is also a bug :(

**Review by MicahZoltu — APPROVED** [review]

**Pull request merged**